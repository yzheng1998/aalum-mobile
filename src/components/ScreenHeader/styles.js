import styled from 'styled-components'
import { themeGet } from 'styled-system'

export const HeaderContainer = styled.View`
  padding-left: 25px;
  padding-right: 25px;
  flex-direction: row;
  align-items: center;
`

export const HeaderText = styled.Text`
  font-family: ${themeGet('fonts.title')};
  font-size: 32px;
  margin-left: 5px;
`

export const ButtonContainer = styled.View`
  flex-grow: 1;
  padding-right: 5px;
  align-items: flex-end;
`
