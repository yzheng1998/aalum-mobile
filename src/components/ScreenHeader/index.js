import React from 'react'
import { HeaderContainer, HeaderText, ButtonContainer } from './styles'
import BackButton from '../BackButton'
import theme from '../../../theme'

const ScreenHeader = ({ navigation, title, showBack, children }) => (
  <HeaderContainer>
    {showBack && (
      <BackButton
        onPress={() => navigation.goBack()}
        color={theme.colors.pureBlack}
        style={{ marginRight: 10 }}
      />
    )}
    <HeaderText>{title}</HeaderText>
    <ButtonContainer>{children}</ButtonContainer>
  </HeaderContainer>
)

export default ScreenHeader
