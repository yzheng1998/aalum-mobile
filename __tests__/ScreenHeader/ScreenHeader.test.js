import React from 'react'
import ScreenHeader from '../../src/components/ScreenHeader'
import styled from 'styled-components'
import { shallow } from 'enzyme'

describe('ScreenHeader', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<ScreenHeader />)
    expect(wrapper).toMatchSnapshot()

    wrapper.setProps({ title: 'title', showBack: true })
    expect(wrapper).toMatchSnapshot()
  })

  test('renders children properly', () => {
    const MockChild = styled.View``
    MockChild.displayName = 'MockChild'
    const wrapper = shallow(
      <ScreenHeader>
        <MockChild />
      </ScreenHeader>
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('MockChild').length).toBe(1)
  })

  //   test('handles goBack correctly', () => {
  //     const goBackMock = jest.fn()
  //     const navigationMock = { goBack: goBackMock }
  //     const wrapper = mountWithTheme(
  //       <ScreenHeader navigation={navigationMock} showBack />
  //     )
  //     const button = wrapper.find('BackButton')
  //     expect(button.length).toBe(1)
  //     button.simulate('press')

  //     expect(goBackMock).toHaveBeenCalledTimes(1)
  //   })
})
