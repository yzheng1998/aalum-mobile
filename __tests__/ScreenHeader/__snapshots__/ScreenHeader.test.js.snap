// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScreenHeader renders children properly 1`] = `
<Styled(View)>
  <Styled(Text) />
  <Styled(View)>
    <MockChild />
  </Styled(View)>
</Styled(View)>
`;

exports[`ScreenHeader renders correctly 1`] = `
<Styled(View)>
  <Styled(Text) />
  <Styled(View) />
</Styled(View)>
`;

exports[`ScreenHeader renders correctly 2`] = `
<Styled(View)>
  <BackButton
    color="#000000"
    onPress={[Function]}
    style={
      Object {
        "marginRight": 10,
      }
    }
  />
  <Styled(Text)>
    title
  </Styled(Text)>
  <Styled(View) />
</Styled(View)>
`;
